### PR TITLE
Added sounds

### PR DESCRIPTION
**Pull request by MMCQ33**
Added sounds and made them scaleable with volume. Added exclamationMark when ant finds resource

### CODE REVIEW

File: Hivemind/Assets/Scripts/Minds/Gathering.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public enum State
     public bool IsScout;
     private int nextHarvest;
     private bool preparingReturn;
-
+    private GameObject excla;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/Minds/Gathering.cs]: Keep this is the function pls, no need for the whole class to know about it

File: Hivemind/Assets/Scripts/TestScript.cs [hunk 2]
```diff
@@ -55,13 +56,16 @@ public void SpawnUnitForLastGroup()
     {
         if (!guid.Equals(Guid.Empty))
         {
-            var go = Instantiate(WorkerPrefab, SpawnPosition.position, Quaternion.identity,
-                GameObject.Find("Ants").transform);
-            var a = go.GetComponent<Ant>();
+            for (int x = 0; x < 20; x++)
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/TestScript.cs]: pls no

File: Hivemind/Assets/Scripts/TestScript.cs [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ﻿using System;
+using System.Security.Cryptography.X509Certificates;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/TestScript.cs]: ????

File: Hivemind/Assets/Scripts/Minds/Gathering.cs [hunk 4]
```diff
@@ -165,11 +172,24 @@ public void Execute(Ant ant)
         }
     }
 
+    private void FoundResource()
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/Minds/Gathering.cs]: Just put everything in the Discover function and start the coroutine where this function is currently called

**Review by ErikLouwerse — CHANGES_REQUESTED** [review]

File: Hivemind/Assets/Scripts/Minds/Gathering.cs [hunk 5]
```diff
@@ -142,7 +142,13 @@ public void Execute(Ant ant)
                     ant.GetAgent().SetDestination(ant.GetStorage().GetPosition());
                     if (ant.AtBase())
                     {
-                        if (IsScout && target != null) target.AddToKnownResourceList();
+                        if (IsScout && target != null)
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/Minds/Gathering.cs]: make stacked if into a single one

File: Hivemind/Assets/Scripts/Minds/Gathering.cs [hunk 6]
```diff
@@ -170,6 +176,14 @@ public double Likelihood(Ant ant)
         return 50;
     }
 
+    private IEnumerator Discover()
+    {
+        ant.PlaySoundDiscovery();
+        var excla = (GameObject)GameObject.Instantiate(Resources.Load("ExclamationMark"), ant.transform, false);
+         excla.transform.localScale *= 3;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/Minds/Gathering.cs]: Spacing

File: Hivemind/Assets/Scripts/ResourceNode.cs [hunk 7]
```diff
@@ -119,4 +138,27 @@ public int GetResourcesFuture()
...
+    {
+        if (audioSrc != null)
+        {
+            if (PlayerPrefs.HasKey("Volume"))
+            {
+                audioVolume = PlayerPrefs.GetFloat("Volume");
+            }
+
+            audioSrc.volume = audioVolume;
+            if (this.resourceType == ResourceType.Crystal)
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/ResourceNode.cs]: useless this.

File: Hivemind/Assets/Scripts/Ant.cs [hunk 8]
```diff
@@ -16,6 +16,8 @@ public class Ant : MonoBehaviour
     public Gathering.State state;
     private Storage storage;
     internal Guid unitGroupID;
+    private AudioSource audioSrc;
+    private float audioVolume = 1f;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/Ant.cs]: Why not just change the audioSrc.volume direly, this field seems redundant

File: Hivemind/Assets/Scripts/ResourceNode.cs [hunk 9]
```diff
@@ -31,6 +35,18 @@ private void Awake()
         futureResourceAmount = resourceAmount;
         gameObject.GetComponent<MeshRenderer>().enabled = false;
         GameWorld.AddNewResource(this);
+        audioSrc = GetComponent<AudioSource>();
+        if (PlayerPrefs.HasKey("Volume"))
+        {
+            audioVolume = PlayerPrefs.GetFloat("Volume");
+        }
+        audioSrc.volume = (audioVolume);
+        if (this.resourceType == ResourceType.Crystal)
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/ResourceNode.cs]: this.

File: Hivemind/Assets/Scripts/ResourceNode.cs [hunk 10]
```diff
@@ -21,6 +21,10 @@ public class ResourceNode : MonoBehaviour
     private int TimeToRespawn = 30;
     public bool DestroyWhenEmpty = false;
 
+    [SerializeField]
+    private AudioSource audioSrc;
+
+    private float audioVolume = 1;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/ResourceNode.cs]: Seems redundant, see other comment

**Review by ErikLouwerse — CHANGES_REQUESTED** [review]

File: Hivemind/Assets/Scripts/Ant.cs (hunk 8 above)
Comment (ErikLouwerse) [reply Hivemind/Assets/Scripts/Ant.cs]: REMOVE PLS

**Review by ErikLouwerse — CHANGES_REQUESTED** [review]

File: Hivemind/Assets/Scripts/Ant.cs [hunk 11]
```diff
@@ -107,9 +116,23 @@ internal void SetStorage(Storage storage)
         this.storage = storage;
     }
 
+    public void UpdateVolume()
+    {
+        if (PlayerPrefs.HasKey("Volume"))
+        {
+            audioSrc.volume = PlayerPrefs.GetFloat("Volume");
+        }
+        audioSrc.volume *=  0.05f;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/Ant.cs]: this needs to be in the if, otherwise the audio will just decrease

**Review by ErikLouwerse — CHANGES_REQUESTED** [review]

**Review by ErikLouwerse — APPROVED** [review]

**Review by zoompower — APPROVED** [review]

**Pull request merged**